services/imagery/test/backends/z-cam-e1-backend.test.js

```diff
@@ -0,0 +1,137 @@
+  const imageFilename = path.join(__dirname, '../fixtures/shape-1.jpg');
+  const image = await fs.readFile(imageFilename, { encoding: null });
+
+  const imageStore = {
+    addImage: jest.fn().mockImplementation(() => Promise.resolve())
+  };
+
+  const t1 = telemetry.CameraTelem.encode({ time: 1, lat: 2 }).finish();
+
+  const cameraApi = nock('http://camera:1234')
+    .defaultReplyHeaders({ 'content-type': 'application/json' })
+    .get('/ctrl/session')
+    .reply(200, { code: 0, desc: '', msg: '' })
+    .get('/ctrl/still?action=single')
+    .reply(200, { code: 0, desc: '', msg: '/DCIM/100MEDIA/EYED6391.JPG' })
+    .get('/DCIM/100MEDIA/EYED6391.JPG')
+    .reply(200, image, { 'content-type': 'image/jpeg' })
+    .get('/DCIM/100MEDIA/EYED6391.JPG?act=rm')
+    .reply(200, { code: 0, desc: '', msg: '' })
+    .get('/ctrl/still?action=single')
+    .reply(200, { code: 0, desc: '', msg: '/DCIM/100MEDIA/EYED6392.JPG' })
+    .get('/DCIM/100MEDIA/EYED6392.JPG')
+    .reply(200, image, { 'content-type': 'image/jpeg' })
+    .get('/DCIM/100MEDIA/EYED6392.JPG?act=rm')
+    .reply(200, { code: 0, desc: '', msg: '' })
+    .get('/ctrl/session?action=quit')
+    .reply(200, { code: 0, desc: '', msg: '' });
+
+  const telemApi = nock('http://telemetry:8081')
+    .defaultReplyHeaders({ 'content-type': 'application/x-protobuf' })
+    .get('/api/camera-telem').reply(200, t1)
+    .get('/api/camera-telem').reply(500);
+
+  const backend = new ZCamE1Backend(
+    imageStore, 500, 'http://camera:1234', 'http://telemetry:8081'
+  );
+  const spy = jest.spyOn(logger, 'error');
+
+  await backend.start();
+  await wait(250);
+
+  try {
+    expect(spy).not.toHaveBeenCalled();
+
+    logger.transports[0].silent = true;
+    await wait(500);
+  } finally {
+    logger.transports[0].silent = false;
+    await backend.stop();
+  }
+
+  cameraApi.done();
+  telemApi.done();
+
+  expect(imageStore.addImage).toBeCalledTimes(2);
+
+  expect(fileType(imageStore.addImage.mock.calls[0][0]).ext).toEqual('jpg');
+  expect(imageStore.addImage.mock.calls[0][1].time).toBeDefined();
+  expect(imageStore.addImage.mock.calls[0][1].has_telem).toBeTruthy();
+  expect(imageStore.addImage.mock.calls[0][1].telem.lat).toEqual(2);
+  expect(fileType(imageStore.addImage.mock.calls[1][0]).ext).toEqual('jpg');
+  expect(imageStore.addImage.mock.calls[1][1].time).toBeDefined();
+  expect(imageStore.addImage.mock.calls[1][1].has_telem).toBeFalsy();
+
+  expect(spy).toHaveBeenCalledTimes(1);
+  expect(spy.mock.calls[0][0]).toMatch(/Error while requesting telemetry/);
+});
+
+test('backend continues after errors', async () => {
+  const cameraApi = nock('http://camera:1234')
+    .defaultReplyHeaders({ 'content-type': 'application/json' })
+    .get('/ctrl/session')
+    .reply(200, { code: 0, desc: '', msg: '' })
+    .get('/ctrl/still?action=single')
+    .reply(500)
+    .get('/ctrl/still?action=single')
+    .reply(200, { code: 1, desc: '', msg: '' })
```

Comment:
Wait, what? The camera has its own error code but still returns 200? ugh..

Reply:
yeah I know, also deleting images with get requests.